File: Data-Structure/loc_is_fun.py
v = [1, 2, 3]

def insert(item: int, loc: int):
    v.append(0)
    for i in range(len(v) - 2, loc - 1, -1):
        v[i + 1] = v[i]

    v[loc] = item

def main():
    item = int(input())
    loc = int(input())

    insert(item, loc)

    for i in range(0, len(v)):
        print(v[i])

main()

File: Data Structure/loc_is_fun.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>
using namespace std;

vector<int> v;

void insert(int item, int loc) {
	v.push_back(0);

	int i;
	for (i = v.size() - 2; i >= loc; i--) {
		v[i + 1] = v[i];
	}

	v[loc] = item;
}

int main() {
	int i;
	for (i = 0; i < 3; i++) v.push_back(i);

	int item;
	cin >> item;

	int loc;
	cin >> loc;

	insert(item, loc);

	for (i = 0; i < v.size(); i++) cout << v[i] << endl;
	return 0;
}